contracts/datafeeds/DataFeedInterface.sol

```diff
@@ -20,18 +17,18 @@ contract DataFeedInterface {
     function getDecimals(address ofAsset) constant returns (uint256) {}
     function getDescriptiveInformation(address ofAsset) constant returns (string, string, uint256, string, bytes32) {}
     function getSpecificInformation(address ofAsset) constant returns (uint256, bytes32, address, address) {}
+    function isValid(address ofAsset) constant returns (bool) {}
+    function getPrice(address ofAsset) constant returns (uint) {}
+    function getInvertedPrice(address ofAsset) constant returns (uint) {}
+    function getReferencePrice(address ofBase, address ofQuote) constant returns (uint) {}
+    function getTimestamp(address ofAsset) constant returns (uint) {}
+    function getData(address ofAsset) constant returns (uint, uint) {}
     // Get data feed specific information
     function getQuoteAsset() constant returns (address) {}
     function getInterval() constant returns (uint) {}
     function getValidity() constant returns (uint) {}
     function getLastUpdateId() constant returns (uint) {}
     function getLastUpdateTimestamp() constant returns (uint) {}
-    // Get asset specific information
-    function getPrice(address ofAsset) constant returns (uint) {}
-    function getInvertedPrice(address ofAsset) constant returns (uint) {}
-    function getReferencePrice(address ofBase, address ofQuote) constant returns (uint) {}
-    function getTimestamp(address ofAsset) constant returns (uint) {}
-    function getData(address ofAsset) constant returns (uint, uint) {}
```

Comment:
the logic of this Interface is:
- Contract A 
   - Contract A **general information**; What differentiates this contract from other contracts w similar bytecode; Eg values passed in as constructor arguments or usage indicators
   - Contract A **specific information**; Eg information about Asset
- Contract B 
   - Contract B **general information**; What differentiates this contract from other contracts w similar bytecode; Eg values passed in as constructor arguments or usage indicators
   - Contract B **specific information**; Eg information about Price

Reply:
ok I can add comments to specify which contracts they fall under; sound reasonable?

Reply:
and move those events back to where they were of couse